packages/dcos-net/extra/dcos-net-setup.py

```diff
@@ -111,7 +111,7 @@ def copy_file(src, dst) -> bool:
         return True
 
 
-def add_config(unit: str, src: str, path: str) -> int:
+def add_config(unit: str, path: str, src: str) -> int:
```

Comment:
from the calls below:

```python
def add_networkmanager_config(src: str) -> int:
    if NETWORK_MANAGER or 'el' in platform.release():
        return add_config('NetworkManager.service', '/etc/NetworkManager/conf.d', src)
    return 0
def add_networkd_config(src: str) -> int:
    # systemd-networkd, when enabled, will wipe the configurations like IP
    # address of network interfaces and this behavior happens only on coreos
    # This problem is tracked by:
    # https://jira.mesosphere.com/browse/DCOS_OSS-1790
    # We need to mark interfaces managed by DC/OS as unmanaged when networkd is
    # enabled on coreos
    if NETWORKD or 'coreos' in platform.release():
        return add_config('systemd-networkd.service', '/etc/systemd/network', src)
```

CI didn't catch this (and neither did I because I ran in on aws ;( ) because the `check_for_unit` is returning that `NetworkManager.service` is not running, so no need to patch it in.

Reply:
Does Flatcar have `coreos` in its release string?

Reply:
Good catch, looks like they changed it as leas as of the most recent ami:

```
core@ip-172-31-25-228 ~ $ docker run python python -c 'import platform; print(platform.release())'
5.4.72-flatcar
```

I've added a commit to also run this on flatcar.